src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -112,7 +112,18 @@ public async Task<ConnectNewHeadersResult> ConsensusTipChangedAsync()
                     this.BestReceivedTip = result.Consumed;
                     this.UpdateBestSentHeader(this.BestReceivedTip);
 
-                    int consumedCount = this.cachedHeaders.IndexOf(result.Consumed.Header) + 1;
+                    uint256 consumedHeaderHash = result.Consumed.Header.GetHash();
+                    int consumedCount = 0;
+
+                    for (int i = this.cachedHeaders.Count - 1; i >= 0; i--)
+                    {
+                        if (this.cachedHeaders[i].GetHash() == consumedHeaderHash)
```

Comment:
Just double check that the headers are cached then this is ok

Reply:
and also as discussed move it to its own method and apply in the `ProcessHeader` method